env0/resource_template.go

```diff
@@ -354,24 +354,36 @@ func templateCreatePayloadFromParameters(prefix string, d *schema.ResourceData)
 }
 
 // Reads template and writes to the resource data.
-func templateRead(template client.Template, d *schema.ResourceData) error {
-	if err := writeResourceData(&template, d); err != nil {
+func templateRead(prefix string, template client.Template, d *schema.ResourceData) error {
+	if err := writeResourceDataEx(prefix, &template, d); err != nil {
 		return fmt.Errorf("schema resource data serialization failed: %v", err)
 	}
 
-	templateReadRetryOnHelper(d, "deploy", template.Retry.OnDeploy)
-	templateReadRetryOnHelper(d, "destroy", template.Retry.OnDestroy)
+	templateReadRetryOnHelper(prefix, d, "deploy", template.Retry.OnDeploy)
+	templateReadRetryOnHelper(prefix, d, "destroy", template.Retry.OnDestroy)
 
 	return nil
 }
 
 // Helpers function for templateRead.
-func templateReadRetryOnHelper(d *schema.ResourceData, retryType string, retryOn *client.TemplateRetryOn) {
-	if retryOn != nil {
-		d.Set("retries_on_"+retryType, retryOn.Times)
-		d.Set("retry_on_"+retryType+"_only_when_matches_regex", retryOn.ErrorRegex)
+func templateReadRetryOnHelper(prefix string, d *schema.ResourceData, retryType string, retryOn *client.TemplateRetryOn) {
+	if prefix != "" {
```

Comment:
ugly... but not much I can do at the moment /:

Reply:
why do we need that prefix?

Reply:
with prefix it's a nested version - type map[string]interface{}. Update the map and then set it back in the schema.
Without a prefix you write directly to the schema with d.Set

Let me know if it's still unclear... the Terraform SDK can get a little clunky.

Reply:
Yea...
maybe `nestedPrefix`? idk for a better name

Reply:
@eranelbaz - the term "prefix" is used in many places throughout the code. I would prefer not to modify it.